test/controllers/omniauth_callbacks_controller_test.rb

```diff
@@ -0,0 +1,21 @@
+require 'test_helper'
+require 'omniauth_helper'
+
+class OmniauthCallbacksControllerTest < ActionController::TestCase
+  before do 
+    request.env["devise.mapping"] = Devise.mappings[:user] 
+    request.env["omniauth.auth"] = OmniAuth.config.mock_auth[:google_oauth2] 
+  end
+
+  describe 'access top page' do
```

Comment:
We should throw this in an integration test rather than a controller test. The code is pretty much spot on, just the wrong spot for it! Will work with you on this.